### PR TITLE
feat: allow adding classes to both modal body (removing extraneous div) and backdrop 

### LINKED ISSUE #800: Modal does not honor provided CSS class in "className" property
### Description
CSS class provided in Modal "className" property does not have any effect.

Currently bypassing this issue by directly manipulating underlying "fundamental-style" CSS property of Modal.

`.fd-modal{
    max-width: none!important;
}
.modal-demo-bg{
    width: 90%;
}`

### Versions
**fundamental-react:** 0.7.1
**fundamental-styles:** 0.3.0

### PR DESCRIPTION
**Pull request by jbadan**
### Description

Copy of https://github.com/SAP/fundamental-react/pull/824, removing commit `01a4c6c` that causes publishing issues.

fixes #800

### CODE REVIEW

File: src/Modal/Modal.js [hunk 1]
```diff
@@ -51,9 +52,14 @@ class Modal extends Component {
...
 
-        const modalClasses = classnames(
+        const backdropClasses = classnames(
             'fd-overlay',
             'fd-overlay--modal',
+            backdropClassName
+        );
+
+        const modalClasses = classnames(
+            'modal-demo-bg',
```
Comment (greg-a-smith) [inline src/Modal/Modal.js]: I know this class was here before, but should it be?  This seems like something used as a selector for unit tests and there is an entry in `customstyles.css` from `fundamental-styles`.  It's fine to use some custom styles for a playground or such, but it shouldn't be part of the base component's classes.

Comment (jbadan) [reply src/Modal/Modal.js]: Good catch! I'll remove it

Comment (jbadan) [reply src/Modal/Modal.js]: @greg-a-smith turns out that entire div was just for the styles docs site. I'm removing it here and going to make a pr there to make that more obvious.

**Comment by greg-a-smith** [comment]
@jbadan Although this _is_ a feature, should the prefix be `fix:` since we are still on version 0?

**Comment by jbadan** [comment]
I did `feat` because I think this technically a breaking change since I removed a div? What do you think?

**Comment by greg-a-smith** [comment]
@jbadan Yeah, fair enough.  I forgot about that. 👍

**Pull request merged**